x/ibc/04-channel/keeper/packet.go

```diff
@@ -396,119 +401,3 @@ func (k Keeper) AcknowledgePacket(
 
 	return packet, nil
 }
-
-// CleanupPacket is called by a module to remove a received packet commitment
-// from storage. The receiving end must have already processed the packet
-// (whether regularly or past timeout).
-//
-// In the ORDERED channel case, CleanupPacket cleans-up a packet on an ordered
-// channel by proving that the packet has been received on the other end.
-//
-// In the UNORDERED channel case, CleanupPacket cleans-up a packet on an
-// unordered channel by proving that the associated acknowledgement has been
-//written.
-func (k Keeper) CleanupPacket(
```

Comment:
So, there is still one reason we might want this function - it's efficiently batch-able in the case of ordered channels, since one verification of the receive sequence is enough to cleanup lots of packets. Maybe that isn't worth the cost of having an extra function for packet cleanup, what do you think?

Reply:
Ah yea, that sounds useful to me. So basically if an application decides not to use acknowledgements, then they would call `CleanupPacket` probably to batch clear an ordered channel. 

I think it is important to note that `CleanupPacket` won't be compatible with using acknowledgements. And going this route will be at the application level since by default the `ibc` module is optimized for the flow of packets to involve eventual acknowledgements.

I'll add back cleanup packet with extra comments and optimized for ordered channel batch clearing